EIPS/eip-signTypedData.md

```diff
@@ -0,0 +1,263 @@
+
+This EIP adds an RPC method to sign arrays of arbitrary typed data, making it easy for the Signer UI to display this data to the user in a human-readable form, so that the user can be 100% confident of exactly what they are signing, while keeping the resulting signatures machine-verifiable. In addition, there is no room for a malicious DApp to provide the signer with alternative data then that being presented to the user, since the signer hashes the typedData before the user signs it.
+
+This method should only be implemented in Ethereum clients with Signer UI's that require user approval before signing (MetaMask, MEW, Parity signer, Ledger, Trezor). It should not be implemented in clients where approval is already granted (e.g. Geth with unlocked accounts).
+
+<img src="https://github.com/0xProject/EIPs/tree/master/EIPS/eip-eth_signTypedData/eth_signTypedData.png" width="500px">
+
+This EIP is a continuation of a discussion here: https://github.com/ethereum/EIPs/pull/683
+I offered this solution in a comment: https://github.com/ethereum/EIPs/pull/683#issuecomment-327945854
+It proposes a more general solution then the one offered here: https://github.com/ethereum/EIPs/pull/693
+
+## Motivation
+
+There are a whole range of higher level protocols emerging on Ethereum. State channels, 0x protocol, Login protocols, etc... For scalability reasons these protocols want to sign some protocol-specific data off-chain. They then use those signed messages to trigger on-chain or other actions with important consequences.
+
+* Transferring ETH
+* Transferring tokens
+* Transferring ownership
+* etc...
+
+
+The current `eth_sign` implementation allows signing of arbitrary data without specifying it's type or structure. Some signers assume that the data is a UTF-8 string. Some signers show it as a hex encoded string. This leads to user confusion since it is impossible to verify the message you're signing unless it's a plaintext UTF-8 string.
+
+<img src="https://github.com/0xProject/EIPs/tree/master/EIPS/eip-eth_signTypedData/eth_sign.png"  width="500px">
+
+Calling `eth_sign` on Metamask displays the string to sign. If the user is signing the result of hashing a complex structure involving multiple critical pieces of information, the only way to verify what they are signing is to re-hash the same structure in an independent script and make sure the hashes match.
+
+Calling `personal_sign` on Metamask with the raw bytes of a hash (e.g. not an ASCII string) shows the user this even less verifiable message that they should sign.
+
+<img src="https://github.com/0xProject/EIPs/tree/master/EIPS/eip-eth_signTypedData/personal_sign.png"  width="500px">
+
+The main problem is that signers don't have enough metadata to display the DApp's real intent effectively. In order to do so, they need the plaintext input that the dApp wishes hashed and signed by the user.
+
+## Specification
```

Comment:
* Please specify the return value of `eth_signTypedData`.
* Please specify the arguments of `eth_signTypedData`.

Reply:
Return value: `signature``
Params: `[signerAddress, typedData]`